frontend/simple/views/containers/proposals-form/Form.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <modal-form
+    :isActive="isActive"
+    :title="L('New Proposal')"
+    :subTitle="activeForm.subTitle"
+    :submitText="L('Start Proposal')"
+    :onSubmit="handleSubmit"
+    :onClose="closeModal"
+  >
+    <component ref="form" :is="activeForm.component"></component>
+  </modal-form>
+</template>
+<script>
+import Vue from 'vue'
+import { ModalForm } from '../../components/Modals/'
+import { mixinL } from '../../utils/mixins'
+import L from '../../utils/translations'
+import Mincome from './Mincome.vue'
+import RuleAddMember from './RuleAddMember.vue'
+import {
+  SHOW_PROPOSAL_CHANGE_MINCOME,
+  SHOW_PROPOSAL_ADD_MEMBER,
+  SHOW_PROPOSAL_REMOVE_MEMBER,
+  SHOW_PROPOSAL_RULE_ADD_MEMBER,
+  SHOW_PROPOSAL_RULE_REMOVE_MEMBER,
+  SHOW_PROPOSAL_RULE_CHANGE_RULE
+} from '../../../utils/events'
+
+export default {
+  name: 'Form',
+  mixins: [
+    mixinL
+  ],
+  components: {
+    ModalForm
+  },
+  data () {
+    return {
+      isActive: false,
+      activeForm: {},
+      forms: {
+        [SHOW_PROPOSAL_CHANGE_MINCOME]: {
+          component: Mincome,
+          subTitle: L('Change minimum income')
+        },
+        [SHOW_PROPOSAL_ADD_MEMBER]: {
+          // TODO on next MR
+        },
+        [SHOW_PROPOSAL_REMOVE_MEMBER]: {
+          // TODO on next MR
+        },
+        [SHOW_PROPOSAL_RULE_ADD_MEMBER]: {
+          component: RuleAddMember,
+          subTitle: L('Change Rule - Add Member')
+        },
+        [SHOW_PROPOSAL_RULE_REMOVE_MEMBER]: {
+          // TODO on next MR
+        },
+        [SHOW_PROPOSAL_RULE_CHANGE_RULE]: {
+          // TODO on next MR
+        }
+      }
+    }
+  },
+  created () {
+    Vue.events.$on(SHOW_PROPOSAL_CHANGE_MINCOME, () => this.openModal(SHOW_PROPOSAL_CHANGE_MINCOME))
+    Vue.events.$on(SHOW_PROPOSAL_RULE_ADD_MEMBER, () => this.openModal(SHOW_PROPOSAL_RULE_ADD_MEMBER))
```

Comment:
Perhaps it would be better to listen to a single event to change the form and pass in the name of the component to be replaced as an argument to the event.

Reply:
A single event I agree, but the argument, should still be a `key`, so all the proposal types information is kept together. Because it's not just the component, it also has the title. Another advantage is when we have a proposal with 2 triggers, the argument is the same `key`, and the info related to that key is kept at `Form.vue`